synapse/module_api/__init__.py

```diff
@@ -1124,6 +1159,26 @@ async def get_room_state(
 
         return {key: state_events[event_id] for key, event_id in state_ids.items()}
 
+    async def defer_to_thread(
+        self,
+        f: Callable[..., TV],
+        *args: Any,
+        **kwargs: Any,
+    ) -> TV:
+        """Runs the given function in a separate thread from Synapse's thread pool.
+
+        Added in Synapse v1.49.0.
+
+        Args:
+            f: The function to run.
+            args: The function's arguments.
+            kwargs: The function's keyword arguments.
+
+        Returns:
+            The return value of the function once ran in a thread.
+        """
+        return await defer_to_thread(self._hs.get_reactor(), f, *args, **kwargs)
+
```

Comment:
While not directly related with the background update work in this PR it's going to be used by https://gitlab.matrix.org/new-vector/ems-synapse-background-update-controller and it's a small change so I thought I'd include it in here. I can appreciate if people think it's too much scope-creeping, and would prefer me to open a separate PR for it.